ledger/ledgercore/statedelta.go

```diff
@@ -69,3 +69,80 @@ type StateDelta struct {
 	// previous block timestamp
 	PrevTimestamp int64
 }
+
+// AccountsDelta stores orderd accounts and allow fast lookup by address
+type AccountsDelta struct {
+	accts      []basics.BalanceRecord
+	acctsCache map[basics.Address]int
+}
+
+// MakeStateDelta creates a new instance of StateDelta
+func MakeStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64) StateDelta {
+	return StateDelta{
+		Accts:         AccountsDelta{},
```

Comment:
allocation the `ad.acctsCache` here would save you from checking it on every `addAccount`.

Reply:
maybe I'm being too greedy here, but initializing the above slice/acctsCache with some size could avoid reallocations down the road.

Reply:
This is done on purpose to allow using non-initialized `AccountsDelta` in acctupdates. I think I'll run benchmarks to double check both options when complete.

Reply:
Try to see if this nil checking can be avoided by design.